inc/TRestGeant4Hits.h

```diff
@@ -20,44 +20,40 @@
 
 #include <TArrayF.h>
 #include <TArrayI.h>
+#include <TObject.h>
 #include <TRestHits.h>
 
 #include <iostream>
 
-#include "TObject.h"
-
 class TRestGeant4Hits : public TRestHits {
    protected:
     TArrayI fVolumeID;
-    TArrayI fProcessID;      // [fNHits]
-    TArrayF fKineticEnergy;  // [fNHits]
+    TArrayI fProcessID;
```

Comment:
why remove `[fNHits]`? This tells ROOT the size of the vector. I believe it is important for I/O streaming.

Reply:
https://root.cern.ch/root/htmldoc/guides/users-guide/InputOutput.html

Reply:
I added it back. I removed it because I wasn't sure it was doing anything and it was not present on all members, so I have added it to all members where it makes sense.

Also I don't understand the usage of classes such as `TArrayF` on `TRestGeant4Hits` while on `TRestHits` we use `std::vector` for the same thing. I think I will replace the root arrays with the `std::vector` for consistency unless anyone has a problem with that.

Reply:
I have removed this root specific comment on c6922a111bff7258818b5f5b9d4273b910c51ec2. I think that this is no longer needed as it is only used on c type arrays on the root documentation, and (to me) it makes no sense to require this information for serialization of a `std::vector` for example, as it is redundant information. It could probably be removed from the base class also, but it would be nice if this was confirmed somehow.